File: challenge-240/pokgopun/go/ch-1.go
//# https://theweeklychallenge.org/blog/perl-weekly-challenge-240/
/*#

Task 1: Acronym

Submitted by: [52]Mohammad S Anwar
     __________________________________________________________________

   You are given two arrays of strings and a check string.

   Write a script to find out if the check string is the acronym of the
   words in the given array.

Example 1

Input: @str = ("Perl", "Python", "Pascal")
       $chk = "ppp"
Output: true

Example 2

Input: @str = ("Perl", "Raku")
       $chk = "rp"
Output: false

Example 3

Input: @str = ("Oracle", "Awk", "C")
       $chk = "oac"
Output: true

Task 2: Build Array
#*/
//# solution by pokgopun@gmail.com

package main

import "fmt"

func main() {
	for _, data := range []struct {
		strings []string
		abbrv   string
		res     bool
	}{
		{[]string{"Perl", "Python", "Pascal"}, "ppp", true},
		{[]string{"Perl", "Raku"}, "rp", false},
		{[]string{"Oracle", "Awk", "C"}, "oac", true},
	} {
		fmt.Println(isAbbrv(data.strings, data.abbrv) == data.res)
	}
}

func isAbbrv(s []string, a string) bool {
	if len(s) != len(a) {
		return false
	}
	for i, v := range s {
		/*
			if (v[0]-a[i])%32 != 0 {
				return false
			}
		*/
		v1, v2 := v[0], a[i]                   // assign 1st char of 1st word and 1st char of abbrv to variables
		for _, e := range [2]*byte{&v1, &v2} { // apply both variable to the same logic
			switch {
			case *e < 65 || *e > 122, *e > 90 && *e < 97: // none alphabetic character is allowed
				return false
			case *e >= 97 && *e <= 122: // convert any lowercase character to uppercase
				*e -= 32
			}
		}
		if v1 != v2 {
			return false
		}
	}
	return true
}

File: challenge-341/pokgopun/perl/ch-1.pl
### https://theweeklychallenge.org/blog/perl-weekly-challenge-341/
=begin

Task 1: Broken Keyboard

Submitted by: [39]Mohammad Sajid Anwar
     __________________________________________________________________

   You are given a string containing English letters only and also you are
   given broken keys.

   Write a script to return the total words in the given sentence can be
   typed completely.

Example 1

Input: $str = 'Hello World', @keys = ('d')
Output: 1

With broken key 'd', we can only type the word 'Hello'.

Example 2

Input: $str = 'apple banana cherry', @keys = ('a', 'e')
Output: 0

Example 3

Input: $str = 'Coding is fun', @keys = ()
Output: 3

No keys broken.

Example 4

Input: $str = 'The Weekly Challenge', @keys = ('a','b')
Output: 3

Example 5

Input: $str = 'Perl and Python', @keys = ('p')
Output: 1

Task 2: Reverse Prefix
=end
### solution by pokgopun@gmail.com
